env_store.go

```diff
@@ -32,13 +32,13 @@ import (
 	"github.com/bank-vaults/secret-init/pkg/provider/vault"
 )
 
-var supportedProviders = []string{
-	file.ProviderName,
-	vault.ProviderName,
-	bao.ProviderName,
-	aws.ProviderName,
-	gcp.ProviderName,
-	azure.ProviderName,
+var ProviderTypes = []provider.Provider{
+	&file.Provider{},
+	&vault.Provider{},
+	&bao.Provider{},
+	&aws.Provider{},
+	&gcp.Provider{},
+	&azure.Provider{},
 }
```

Comment:
we can register provider factories here and create them elsewhere

```suggestion
var factories = []provider.ProviderFactory{
	file.NewProvider,
	vault.NewProvider,
	// and the rest
}
```

since all provider operations are handled through env store, we can create providers there on demand in store constructor, for example

```go
type EnvStore struct {
	data      map[string]string
	providers []provider.Provider
}

func NewEnvStore(ctx context.Context, config *common.Config) *EnvStore {
	// load env data
	environ := make(map[string]string, len(os.Environ()))
	for _, env := range os.Environ() {
		split := strings.SplitN(env, "=", 2)
		name := split[0]
		value := split[1]
		environ[name] = value
	}

	// create providers
	var providers []provider.Provider
	for _, factory := range factories {
		p, err := factory(ctx, config)
		if err != nil {
			return nil, fmt.Errorf("failed to create provider from factory: %w", err)
		}

		providers = append(providers, p)
	}

	return &EnvStore{
		data:      environ,
		providers: providers,
	}
}
```
You can now simply use provider slice to streamline all operations (including parallelization ops)